documentation/modules/ROOT/pages/connectors/postgresql.adoc

```diff
@@ -2256,6 +2256,94 @@ Again, regularly emitting events solves the problem.
+For example, when a database server that a connector monitors stops or crashes, after the connector re-establishes communication with the PostgreSQL server, it continues to read from the last position recorded by the log sequence number (LSN) offset.
+The connector obtains the information about the last recorded offset from the write-ahead log (WAL) at the configured PostgreSQL replication slot.
+
+However, during a PostgreSQL upgrade, all replication slots are removed, and these slots are not recreated automatically.
+As a result, when the connector attempts to read the last offset from the WAL files, it cannot find the information, and the connector fails with an error.
+To avoid connector failures, and to ensure that no data is lost, you must follow specific steps before and after the upgrade.
+
+The following procedure lists the steps for performing a PostgreSQL database upgrade so that {prodname} continues to capture events, and no data loss occurs:
+
+.Procedure
+
+1. Ensure that no data changes can occur during the upgrade process by temporarily stopping applications that write to the database, or putting them into a read-only mode.
+
+2. Back up the database.
+
+3. Temporarily disable write access to the database.
+
+4. Verify that any changes that occured in the database before you blocked write operations are saved to the WAL on the replication slot.
+
+5. Provide the connector with enough time to capture all event records that are written to the replication slot. +
+This step ensures that all change events that occured before the downtime are accounted for, and that they are saved to Kafka.
+
+6. Verify that the connector has finished consuming entries from the replication slot by checking the value of the flushed LSN.
+
+7. Shut down the connector gracefully by stopping Kafka Connect. +
+Kafka Connect stops the connectors, flushes all event records to Kafka, and records the last offset received from each connector.
+//   Do we need to delete the connector and its offset topic?
+
+8. As a PostgreSQL administrator, drop the replication slot on the primary database server.
+//   Can this be done via setting xref:postgresql-property-slot-drop-on-stop[`slot.drop.on.stop`] to `true`?
+
+9. Stop the database.
+
+10. Perform the upgrade using an approved PostgreSQL upgrade procedure, such as `pg_upgrade` or `pg_dump/restore`.
+
+11. (Optional) Use a standard Kafka tool to remove the connector offsets from the offset storage topic. +
+For an example of how to remove connector offsets, see https://debezium.io/documentation/faq/#how_to_remove_committed_offsets_for_a_connector[how to remove connector offsets] in the {prodname} community FAQ.
+
+12. Restart the database.
+
+13. As a PostgreSQL administrator, create a {prodname} logical replication slot on the database.
+You must create the slot before enabling writes to the database.
+Otherwise, {prodname} cannot capture the changes, resulting in data loss.
```

Comment:
See above, there are three sitiations that can happen

* slot is not created and auto creation of slot enabled at Debezium config - probable data loss
* slot is not created and auto creation of slot disabled at Debezium config - connector refuses to start
* slot is created but old LSN stored in connector offsets - undefined behaviour, data loss